libethcore/ChainOperationParams.h

```diff
@@ -77,10 +77,13 @@ struct ChainOperationParams
 	/// General chain params.
 private:
 	u256 m_blockReward;
+	bool m_allowRewardOverwrite; //allow overwrite of the block reward that is set in genesis.json
```

Comment:
Why do you need this hack?

Reply:
without this, Byzantium blockreward is reset to non zero value even if I want to run Byzantium tests with 0 mining reward that I set in chain params.

Reply:
You're not running Byzantium rules then. Maybe create another config similar to Byzantium but without block reward?

Reply:
You need to create new `EVMSchedule` in `EVMSchedule.h`, add it to `ChainOperationParams::scheduleForBlockNumber()`, then use new param like "byzantiumTestForkBlock" in actual chain config

Reply:
isn't it too much just for the test purpose?

Reply:
Well ad-hoc hacking lower-level data structures and mechanisms just for the test purpose seems too much to me

Reply:
it is not really a hack. its an option to set mining reward in genesis config

Reply:
To make it work you could also try just overriding `NoProof::blockReward()` and always returning 0 from there.

Reply:
Though I wouldn't fully support such change, because it makes `NoProof` engine further away from real network mechanics and for unit tests I would like it to stay closer. But if you'd made another engine just for your tests like `NoProofZeroRewards` that would be fine I guess.